services/s3/src/main/java/software/amazon/awssdk/services/s3/checksums/ChecksumsEnabledValidator.java

```diff
@@ -61,15 +62,18 @@ private ChecksumsEnabledValidator() {
     }
 
     /**
-     * Checks if trailing checksum is enabled for {@link S3Client#getObject(GetObjectRequest)} per request.
+     * Checks if trailing checksum is enabled and {@link ChecksumMode} is disabled for
+     * {@link S3Client#getObject(GetObjectRequest)} per request.
      *
      * @param request the request
      * @param executionAttributes the executionAttributes
-     * @return true if trailing checksums is enabled, false otherwise
+     * @return true if trailing checksums is enabled and ChecksumMode is disabled, false otherwise
      */
-    public static boolean getObjectChecksumEnabledPerRequest(SdkRequest request,
-                                                             ExecutionAttributes executionAttributes) {
-        return request instanceof GetObjectRequest && checksumEnabledPerConfig(executionAttributes);
+    public static boolean getObjectChecksumValidationEnabledChecksumModeDisabled(SdkRequest request,
```

Comment:
lol, this is a bit mouthful... Can we keep the original name and add more clarification in the Javadoc?